packages/server/src/time-control.ts

```diff
@@ -40,100 +54,326 @@ export function ValidateTimeControlBase(
+      default: {
+        console.error("received config with invalid time control type");
+      }
+    }
 
-        if (timeData.forPlayer[playerNr] === null) {
-          console.error(`game with id ${game.id} has defect time control data`);
-          return;
-        }
+    return timeControl;
+  }
 
-        const playerData = timeData.forPlayer[playerNr];
+  handleMove(
+    game: GameResponse,
+    game_obj: AbstractGame<unknown, unknown>,
+    playerNr: number,
+  ): ITimeControlBase {
+    const config = game.config as IConfigWithTimeControl;
 
+    switch (config.time_control.type) {
+      case TimeControlType.Absolute: {
+        let timeControl = game.time_control;
+        // could happen for old games
+        if (timeControl === undefined) {
+          timeControl = this.initialState(game.variant, config);
+        }
+        const playerData = timeControl.forPlayer[playerNr];
         const timestamp = new Date();
 
-        timeData.moveTimestamps.push(timestamp);
-        if (playerData.onThePlaySince !== undefined) {
-          // with this construction, time will not be substracted before the first move of a player
-          // which is good imho
+        timeControl.moveTimestamps.push(timestamp);
+
+        if (playerData.onThePlaySince !== null) {
           playerData.remainingTimeMS -=
             timestamp.getTime() - playerData.onThePlaySince.getTime();
         }
         playerData.onThePlaySince = null;
+        this._timeoutService.clearPlayerTimeout(game.id, playerNr);
+
+        // time control starts only after the first move of player
+        const nextPlayers = game_obj
+          .nextToPlay()
+          .filter((playerNr) => game.moves.some((move) => playerNr in move));
         nextPlayers.forEach((player) => {
-          if (
-            timeData.forPlayer[player] &&
-            timeData.forPlayer[player].remainingTimeMS
-          ) {
-            timeData.forPlayer[player].onThePlaySince = timestamp;
-          }
+          timeControl.forPlayer[player].onThePlaySince = timestamp;
+          this._timeoutService.scheduleTimeout(
+            game.id,
+            player,
+            timeControl.forPlayer[player].remainingTimeMS,
+          );
         });
 
-        await gamesCollection()
-          .updateOne(
-            { _id: new ObjectId(game.id) },
-            { $set: { time_control: timeData } },
-          )
-          .catch(console.log);
-        break;
+        return timeControl;
       }
 
       case TimeControlType.Invalid:
+        throw Error(`game with id ${game.id} has invalid time control type`);
+    }
+  }
+
+  getMsUntilTimeout(game: GameResponse, playerNr: number): number | null {
+    if (!HasTimeControlConfig(game.config)) {
+      console.log("has no time control config");
+
+      return null;
+    }
+
+    switch (game.config.time_control.type) {
+      case TimeControlType.Absolute:
+      case TimeControlType.Fischer: {
```

Comment:
Does the Fischer stuff work now? Any reason not to enable it in [TimeControlConfigForm.vue](https://github.com/govariantsteam/govariants/blob/40a7258a5a6eb8ddd6ea2de1d0b9a90c64aa4282/packages/vue-client/src/components/GameCreation/TimeControlConfigForm.vue#L38)?  How do we test it?

Reply:
It does not work yet, but I believe that in this specific function (getMsUntilTimeout) it can use the same code block.